app/cni/pkg/install/installer_config.go

```diff
@@ -48,136 +52,179 @@ func (i InstallerConfig) Validate() error {
 	return nil
 }
 
-func findCniConfFile(mountedCNINetDir string) (string, error) {
-	matches, err := filepath.Glob(mountedCNINetDir + "/*.conf")
-	if err != nil {
-		return "", err
+func (c *InstallerConfig) PostProcess() error {
+	if c.CniConfName != "" {
+		return nil
 	}
 
-	file, found := lookForValidConfig(matches, isValidConfFile)
-	if found {
-		return filepath.Base(file), nil
-	}
+	for _, ext := range []string{"*.conf", "*.conflist"} {
+		matches, err := filepath.Glob(filepath.Join(c.MountedCniNetDir, ext))
+		if err != nil {
+			log.Info("failed to search for CNI config files", "error", err)
+			continue
+		}
 
-	matches, err = filepath.Glob(mountedCNINetDir + "/*.conflist")
-	if err != nil {
-		return "", err
-	}
-	file, found = lookForValidConfig(matches, isValidConflistFile)
-	if found {
-		return filepath.Base(file), nil
+		if file, ok := lookForValidConfig(matches, isValidConfFile); ok {
+			log.Info("found CNI config file", "file", file)
+			c.CniConfName = filepath.Base(file)
```

Comment:
Changing/overriding a configuration value in the app logic seems not a  good idea. Can we use a return value?

Reply:
Thanks for noting this. If you agree it's not a good idea to changing the configuration value, could you also improve this logic since you are doing a refactoring

Reply:
That's exactly where it should happen. We're using general logic to load env vars into shared configurations: https://github.com/kumahq/kuma/blob/c3afa4a4509998c242a64d2b575a90a175f913c0/pkg/config/config.go#L8  
I'd agree with you if we were explicitly creating an `InstallerConfig` instance, but that's not the case here.

Reply:
Thanks for explaning and I'll mark as resolved since we are only making default values to this config item